system-contracts/test/NonceHolder.spec.ts

```diff
@@ -0,0 +1,242 @@
+import { expect } from "chai";
+import type { NonceHolder } from "../typechain";
+import { NonceHolderFactory } from "../typechain";
+import {
+  TEST_BOOTLOADER_FORMAL_ADDRESS,
+  TEST_DEPLOYER_SYSTEM_CONTRACT_ADDRESS,
+  TEST_NONCE_HOLDER_SYSTEM_CONTRACT_ADDRESS,
+} from "./shared/constants";
+import { prepareEnvironment, setResult } from "./shared/mocks";
+import { deployContractOnAddress, getWallets } from "./shared/utils";
+import { ethers, network } from "hardhat";
+import { BigNumber } from "ethers";
+
+describe("NonceHolder tests", () => {
+  const wallet = getWallets()[0];
+  let nonceHolder: NonceHolder;
+  let nonceHolderAccount: ethers.Signer;
+  let systemAccount: ethers.Signer;
+  let deployerAccount: ethers.Signer;
+
+  before(async () => {
+    await prepareEnvironment();
+    await deployContractOnAddress(await wallet.getAddress(), "NonceHolder");
+    await deployContractOnAddress(TEST_NONCE_HOLDER_SYSTEM_CONTRACT_ADDRESS, "NonceHolder");
+    nonceHolder = NonceHolderFactory.connect(await wallet.getAddress(), wallet);
+    nonceHolderAccount = await ethers.getImpersonatedSigner(await wallet.getAddress());
+    systemAccount = await ethers.getImpersonatedSigner("0x000000000000000000000000000000000000900b");
```

Comment:
Btw, I think this value is already defined in the constants

Reply:
done